lib/connectors/mongodb/connector.rb

```diff
@@ -53,55 +53,61 @@ def initialize(configuration: {})
         @user = configuration.dig(:user, :value)
         @password = configuration.dig(:password, :value)
         @direct_connection = configuration.dig(:direct_connection, :value)
+
+        if $gclient_cache.key?(@host) 
```

Comment:
Init 1 client per host (bugs with settings changes potentially, but that's for illustration purely)